src/main/java/org/jabref/logic/journals/PredatoryJournalLoader.java

```diff
@@ -0,0 +1,136 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PredatoryJournalLoader {
+    private static class PJSource {
+        URL url = null;
+        final Pattern elementPattern;
+
+        PJSource(String url, String regex) {
+            try {
+                this.url = new URL(url);
+            } catch (MalformedURLException ex) {
+                LOGGER.error("Malformed URL has occurred in PJSource", ex);
+            }
+            this.elementPattern = (regex != null) ? Pattern.compile(regex) : null;
+        }
+    }
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(PredatoryJournalLoader.class);
+    private static final Pattern PATTERN_NAME = Pattern.compile("(?<=\">).*?(?=<)");
+    private static final Pattern PATTERN_URL = Pattern.compile("http.*?(?=\")");
+    private static final Pattern PATTERN_ABBR = Pattern.compile("(?<=\\()[^ ]*(?=\\))");
+    private static final List<String> LINK_ELEMENTS = new ArrayList<>();
+    private static final List<PJSource> PREDATORY_SOURCES = List.of(
+            new PJSource("https://raw.githubusercontent.com/stop-predatory-journals/stop-predatory-journals.github.io/master/_data/journals.csv",
+                    null),
+            new PJSource("https://raw.githubusercontent.com/stop-predatory-journals/stop-predatory-journals.github.io/master/_data/publishers.csv",
+                    null),
+            new PJSource("https://beallslist.net/",
+                    "<li>.*?</li>"),
+            new PJSource("https://beallslist.net/standalone-journals/",
+                    "<li>.*?</li>"),
+            new PJSource("https://beallslist.net/hijacked-journals/",
+                    "<tr>.*?</tr>")
+    );
+    private PredatoryJournalRepository repository = new PredatoryJournalRepository();
+
+    public static PredatoryJournalRepository loadRepository() {
+        PredatoryJournalLoader loader = new PredatoryJournalLoader();
+        // Initialize with built-in list
+        try (InputStream resourceAsStream = JournalAbbreviationRepository.class.getResourceAsStream("/journals/predatoryJournalList.mv")) {
+            if (resourceAsStream == null) {
+                LOGGER.warn("There is no predatoryJournalList.mv. We use a default predatory journal list");
+            } else {
+                Path tempDir = Files.createTempDirectory("jabref-journal");
+                Path tempJournalList = tempDir.resolve("predatoryJournalList.mv");
+                Files.copy(resourceAsStream, tempJournalList);
+                loader.repository = new PredatoryJournalRepository(tempJournalList);
+                tempDir.toFile().deleteOnExit();
+                tempJournalList.toFile().deleteOnExit();
+            }
+        } catch (IOException e) {
+            LOGGER.error("Error while copying predatory journal list", e);
+            return null;
+        }
+
+        loader.update();
+        return loader.repository;
+    }
+
+    public void update() {
```

Comment:
Please find a way to execute the updating in a background thread: BackgroundTask.wrap() -> .... otherwise it will block the main javafx thread on the start and will lead to freezing until the files are downloaded.

Reply:
Yep I have noticed that as well. I will work on it now